mypy/newsemanal/semanal.py

```diff
@@ -2338,7 +2338,15 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
                 self.progress = True
                 # We need to defer so that this change can get propagated to base classes.
                 self.defer()
-            existing.node = alias_node
+            if isinstance(existing.node, TypeAlias):
```

Comment:
What exactly goes wrong without this?

Reply:
In the test case the alias initially has `<placeholder C>` nested inside, it is deferred and re-analyzed to correctly have `<instance C>`. But before that, the importer already bounds name to the old (partially analyzed) alias node. Therefore we need to update existing node instead of creating a new `TypeAlias` node.

Reply:
Then this causes a crash because the placeholder type leaks out of semantic analysis.